### DISCUSSION TITLE
how to work together with nvchad?

### DISCUSSION THREAD

**Discussion by twmht** [Exafunction/codeium.nvim#148] [General]
> If you are seeing the codeium source as unused in :CmpStatus, make sure that nvim-cmp setup happens before the codeium.nvim setup.

I have no idea how to confiugre vim-cmp setup happens before the codeium.nvim setup.

Any idea?

**Comment by aquilesg** [comment]
Here's how I did it. I had to modify the following in the ./lua/custom/ directory.

Add the `nvim-cmp` as an override like I did here:
https://github.com/aquilesg/nvchadconfig/blob/main/plugins.lua#L32-L44

Then create an entry within the ./lua/custom/configs/overrides.lua file like so:
https://github.com/aquilesg/nvchadconfig/blob/main/configs/overrides.lua#L77-L119

**Comment by twmht** [comment]
@aquilesg 

yup it works, thank you.

but i can't let codeium generate the code like they do in the gif (https://user-images.githubusercontent.com/1908017/213154744-984b73de-9873-4b85-998f-799d92b28eec.gif). 

and the most powerful utility is that you can generate the code based on your comment, but i still can't use this feature as well.

do you have any idea?

**Comment by aquilesg** [comment]
Hmm it looks like it works on my end?
![Screenshot 2024-02-23 at 9 25 24 PM](https://github.com/Exafunction/codeium.nvim/assets/20983181/1279ca2b-9769-4ab6-b9b2-b10143528dd4)

**Comment by twmht** [comment]
@aquilesg 

I downgrade to nvim 0.9 and got better experience, which version of nvim you used? btw, why you did this https://github.com/aquilesg/nvchadconfig/blob/main/configs/overrides.lua#L95? I thought "< CR >" is more intutive to select the suggession, isnt it?

**Reply by aquilesg** [reply]
You can take it off it's preference.

**Reply by aquilesg** [reply]
This is the only bit you really need
https://github.com/aquilesg/nvchadconfig/blob/main/configs/overrides.lua#L80-L88

**Comment by twmht** [comment]
@aquilesg 

thanks for your sharing, it really saved my day!

Here is one more question, how can i trigger codeium suggesstions without typing any prefix word. because in many cases i want to write a comment and wait unitl codeium shows its suggesstions, and i can use < C-g > to accept that. but in cmp i need to type some prefix.

do you have any idea?

**Reply by aquilesg** [reply]
That I don't know. I think you can probably configure that via nvim-cmp though.
